cdap-api/src/main/java/co/cask/cdap/api/dataset/table/RowColumns.java

```diff
@@ -43,16 +48,19 @@
 
   public RowColumns(byte[] row) {
     this.row = row;
-    this.columns = new ArrayList<>();
   }
 
   public RowColumns(byte[] row, byte[]... columns) {
     this(row);
+    this.columns = new ArrayList<>();
     add(columns);
   }
 
   @SuppressWarnings("unchecked")
   public T add(byte[]... columns) {
+    if (this.columns == null) {
```

Comment:
But it is possible that someone computes a set of rows and wants to add that. That would be awkward with the suggested API. (Our own IndexedTable does that, right?)